### PR TITLE
fix: ensure statusCode is always an integer

### PR DESCRIPTION
**Pull request by Uzlopak**
I have some bug, where the status code is set to a string value. One way to fix it, is by enforcing status to be always an integer.

### CODE REVIEW

**Comment by Uzlopak** [comment]
@gr2m 
@wolfy1339 

could you consider reviewing this please?

**Comment by wolfy1339** [comment]
This only changes it on the error. Should there not be a middleware in `@octokit/request` instead

**Comment by Uzlopak** [comment]
@wolfy1339 

I am still investigating on the original issue. But this can be make it more resilient anyway. So I would still recommend the PR.

File: src/index.ts [hunk 1]
```diff
@@ -36,7 +36,15 @@ export class RequestError extends Error {
...
 
     this.name = "HttpError";
-    this.status = statusCode;
+
+    // Implicit coercion to number if statusCode is a string
+    this.status = Number.parseInt(statusCode as unknown as string);
+
+    // If status is not equal to itself, then it is NaN
+    // we set then the status to 0
+    if (this.status !== this.status) {
```
Comment (gr2m) [inline src/index.ts:45]: why not use `isNaN()`

```suggestion
    if (isNan(this.status)) {
```

**Review by gr2m — APPROVED** [review]
I'm okay with it, just would use `isNan`

File: src/index.ts (hunk 1 above)
Comment (Uzlopak) [reply src/index.ts:45]: its actually quite faster but i am ok. 

```suggestion
    if (Number.isNaN(this.status)) {
```

**Review by wolfy1339 — APPROVED** [review]

**Comment by Uzlopak** [comment]
@wolfy1339 
@gr2m 

Can we please merge it? THen I could check if this workaround works on gcf. :)

**Pull request merged**